include/main.h

```diff
@@ -0,0 +1,19 @@
+#ifndef MAIN_H
+#define MAIN_H
+
+#include "ultra64.h"
+
+#include "stdint.h"
+
+extern s32 gScreenWidth;
+extern s32 gScreenHeight;
+extern u32 gSystemHeapSize;
+
+extern uintptr_t gSegments[NUM_SEGMENTS];
+extern OSThread gGraphThread;
```

Comment:
actually, if we put `gSegments` after `gAppNmiBufferPtr` in the above we can probably get rid of the first `increment_block_number` (still would like to not add any more though)

Reply:
do changes still need to be made for this?

Reply:
cadmic will have to tell us if the current PR (which received a few updates since the comments) is now ok on that front

I did just remove `gGraphThread` from main.h, it was a faulty copypaste from MM where it is (and needs to be) global.

Reply:
seems fine now, the essay was because I thought I need to fight for `sGraphThread` but never mind that if it's just a copy-paste error